x-pack/winlogbeat/_meta/config/output-elasticsearch.yml.tmpl

```diff
@@ -0,0 +1,15 @@
+{{subheader "Elasticsearch Output"}}
+output.elasticsearch:
+  # Array of hosts to connect to.
+  hosts: ["localhost:9200"]
+
+  # Protocol - either `http` (default) or `https`.
+  #protocol: "https"
+
+  # Authentication credentials - either API key or username/password.
+  #api_key: "id:api_key"
+  #username: "elastic"
+  #password: "changeme"
+
+  # Pipeline to route events to security, sysmon or powershell pipelines
+  pipeline: "winlogbeat-%{[agent.version]}-default-default-routing"
```

Comment:
> default-default

That looks odd. Could we remove those and call it `winlogbeat-8.0.0-routing`?

Reply:
I don't like current naming either.

If we change it, we need to do a deeper refactor in the filebeat/fileset code so that pipelines can have different names.  Right now the code expects

beat_name - version - module_name - fileset_name - filename

what if we switched to

winlogbeat-%{[agent.version]}-x_pack-builtin-routing

that would be more descriptive.  I'm OK with refactoring the filebeat/fileset code too, but that opens up more opportunity for accidentally changing behavior.

Reply:
I haven't looked at the filebeat implementation of the method that generates the name, but based on what it does I think refactoring to allow `""` skipping empty strings sounds like low risk. Like append all the non-empty parts to a `[]string` then join the strings with a `-`.